src/Lachain.Core/RPC/HTTP/FastSyncService.cs

```diff
@@ -0,0 +1,536 @@
+
+        private Dictionary<ulong, Tuple<List<ulong>, List<byte[]>>> _repoBlocks =
+            new Dictionary<ulong, Tuple<List<ulong>, List<byte[]>>>();
+
+        Dictionary<ulong, List<ulong>> _blockVersion = new Dictionary<ulong, List<ulong>>();
+
+        private List<ulong> _repoList = new List<ulong>()
+        {
+            1, 4, 5, 6, 7, 9, 10
+        };
+
+        private bool _fastSyncStatus = false;
+
+        public FastSyncService(
+            IStateManager stateManager,
+            IBlockManager blockManager,
+            IRocksDbContext rocksDbContext,
+            IBlockSynchronizer blockSynchronizer,
+            IConfigManager configManager
+        )
+        {
+            _stateManager = stateManager;
+            _blockManager = blockManager;
+            _rocksDbContext = rocksDbContext;
+            _blockSynchronizer = blockSynchronizer;
+
+            _nodeRepository = new NodeRepository(_rocksDbContext);
+            _versionRepository = new VersionRepository(_rocksDbContext);
+
+            _configManager = configManager;
+        }
+
+        private void GetSerializedNodesAndIds(ulong repoType)
+        {
+            Logger.LogTrace($"Start: GetSerializedNodesAndIds");
+
+            ulong version = 0;
+
+            switch (repoType)
+            {
+                case 1:
+                    version = _stateManager.CurrentSnapshot.Balances.Version;
+                    break;
+                case 4:
+                    version = _stateManager.CurrentSnapshot.Contracts.Version;
+                    break;
+                case 5:
+                    version = _stateManager.CurrentSnapshot.Storage.Version;
+                    break;
+                case 6:
+                    version = _stateManager.CurrentSnapshot.Transactions.Version;
+                    break;
+                case 7:
+                    version = _stateManager.CurrentSnapshot.Blocks.Version;
+                    break;
+                case 9:
+                    version = _stateManager.CurrentSnapshot.Events.Version;
+                    break;
+                case 10:
+                    version = _stateManager.CurrentSnapshot.Validators.Version;
+                    break;
+                default:
+                    break;
+            }
+
+            var versionFactory = new VersionFactory(_versionRepository.GetVersion(Convert.ToUInt32(repoType)));
+            var trieHashMap = new TrieHashMap(_nodeRepository, versionFactory);
+
+            if (trieHashMap.GetNodeIds(version).ToList().Count > 0)
+            {
+                var t = Tuple.Create(trieHashMap.GetNodeIds(version).ToList(),
+                    trieHashMap.GetSerializedNodes(version).ToList());
+
+                _repoBlocks.Add(repoType, t);
+            }
+
+            Logger.LogTrace($"End: getSerializedNodesAndIds");
+        }
+
+        [JsonRpcMethod("getBlocks")]
+        private JObject? GetBlocks(ulong repoType, int offset)
```

Comment:
Looks like this implementation can cause issues if several clients call it simultaneously

Reply:
there would be only 1 client call at the time and won't be any simultaneous calls. this has been taken care by the `handShake` mehod